Update make-a-plant-watering-robot.md (#2329)
viamrobotics/docs: docs/tutorials/projects/make-a-plant-watering-robot.md

## Set up your plant watering robot

Before programming the Pi to make the plant watering robot functional, you need to physically set up the plant watering robot by wiring the different components together.

> You will set up the robot to receive signals from the resistive soil moisture sensor and signal to the peristaltic pump when it is time to pump water from the water container to the plant's container.

Reviewer: ```suggestion
You will set up the robot to receive signals from the resistive soil moisture sensor and signal to the peristaltic pump when it is time to pump water from the water container to the plant container.
```

{{% /alert %}}

> ## Get ready to program your plant watering robot

Reviewer: I don’t think this change makes sense since adding the code etc. is all within this heading. If you want to make this change, I suggest adding another H2 heading later for actually programming it.

Author: i will still add a section with full code so i will apply this comment when i do that change if that's ok with you

Reviewer: Okay I won't block for now then. When you do change the H2 headings, I'd suggest "Set up your hardware" for the first one instead of "Set up your plant watering robot," and then this one can become "Set up your software" or something like that, and then "Program your plant watering robot" can come a couple steps down from here.

Finally, select **Finish**.
Restart your Pi using `sudo reboot` to make these changes take effect.

### Install the Python SDK

Next, set up a virtual environment and install the SDK. (If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).) 

Make sure any packages on your Pi are up to date, while connected to your Pi with SSH run:

```shell
sudo apt update
sudo apt upgrade
```

Then run the following command to create and activate the virtual environment:

> python3 -m venv .venv
> source .venv/bin/activate

Reviewer 2: ```suggestion
```sh {class="command-line" data-prompt="$"}
python3 -m venv .venv
source .venv/bin/activate
```
```

Reviewer 2: also needs a closing code ```

Finally, select **Finish**.
Restart your Pi using `sudo reboot` to make these changes take effect.

### Install the Python SDK

> Next, set up a virtual environment and install the SDK. (If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).) 

Reviewer: ```suggestion
Next, set up a virtual environment and install the SDK.
(If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).) 
```
New line per sentence

Make sure any packages on your Pi are up to date, while connected to your Pi with SSH run:

```shell
sudo apt update
sudo apt upgrade
```

Then run the following command to create and activate the virtual environment:

```sh {class="command-line" data-prompt="$"}
python3 -m venv .venv

> source .venv/bin/activate

Author: ```suggestion
source .venv/bin/activate
```
```

> 

Reviewer: ```suggestion
```

```

Finally, select **Finish**.
Restart your Pi using `sudo reboot` to make these changes take effect.

### Install the Python SDK

Next, set up a virtual environment and install the SDK.
(If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).) 

Make sure any packages on your Pi are up to date, while connected to your Pi with SSH run:

```shell
sudo apt update
sudo apt upgrade
```

Then run the following command to create and activate the virtual environment:

```sh {class="command-line" data-prompt="$"}
python3 -m venv .venv

> source .venv/bin/activate

Author: ```suggestion
source .venv/bin/activate
```
```

Finally, select **Finish**.
Restart your Pi using `sudo reboot` to make these changes take effect.

> ### Install the Python SDK

Reviewer 3: Please move this to be in the section where the user adds control code and runs that code.  It feels out of place here since they don't actually run any code, so let's at this point continue with the component setup and testing

> Next, set up a virtual environment and install the SDK.
> (If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).)

Reviewer 2: Parentheses should be rarely used. It's ok occasionally to list examples but in this case there is no need
```suggestion
Next, create a virtual environment and install the SDK.
If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).
```

> (If you want to read more on virtual environments, check out [the documentation](https://docs.viam.com/build/program/python-venv/).)

Reviewer 2: However this is also out of context and would be better placed alongside the commands you're runnign to create the virtual environment (lines 185+)

> Make sure any packages on your Pi are up to date, while connected to your Pi with SSH run:

Reviewer 2: Let's explicitly connect the action to what they're doing:
```suggestion
Make sure any packages on your Pi are up to date by running the following commands while connected to your Pi with SSH:
```

sudo apt install python3-pip
```

> Run the following command while connected to your Pi with SSH to install [`Adafruit_CircuitPython_MCP3xxx`](https://github.com/adafruit/Adafruit_CircuitPython_MCP3xxx/):

Reviewer 2: Why don't we want people to do this first for testing?

sudo apt install python3-pip
```

Run the following command to install the SDK:

> ```shell

Reviewer 2: ```suggestion
```sh {class="command-line" data-prompt="$"}
```

sudo apt install python3-pip
```

Run the following command to install the SDK:

```sh {class="command-line" data-prompt="$"}
pip3 install viam-sdk
```

> ### Configure the components of your robot in the Viam app

Reviewer 3: This section is missing instructions on how to test the ADC module from the control tab.  We should have them do this so they can understand how to test a sensor from the control tab - which replaces using the external adafruit library to test(which you are correct in removing)

   {{% snippet "show-secret.md" %}}

> 4. Paste this code sample into a new file in the virtual directory you created on your Pi.

Reviewer 2: what's this directory supposed to be? previously it was a directory you instructed a user to create

Author: the virtual environment. since they are not testing they don't need a specific directory, they will only have one script

Reviewer 2: then you can't call that a virtual directory

```suggestion
4. Paste this code sample into a new file on your Pi.
```

    readings = await sensor.get_readings()
    soil_moisture = readings.get('moisture')

>     # Calculate the average moisture reading from the list of readings, to account

Author: ```suggestion
    # Calculate the average moisture reading from the list of readings,
```

>     # for outliers

Author: ```suggestion
    #  to account for outliers
```

    readings = await sensor.get_readings()
    soil_moisture = readings.get('moisture')

>     # Calculate the average moisture reading from the list of readings,

Reviewer: ```suggestion
    # Calculate the average moisture reading from the list of readings
```
[nit]

```

To tinker this example code to work best for you, determine at what [analog value from the soil moisture readings](#test-your-soil-moisture-readings-on-your-pi) you want to water your plant, as your thirsty plant's average moisture reading might differ from our example value of `60000`.

> Test your sensor by putting it in air, water, and different soils to see how the values change to determine your baseline for wet and dry values.

Reviewer: ```suggestion
Test your sensor by putting it in air, water, and different soils to see how the values change and determine your baseline for wet and dry values.
```
Instead of “to” twice

```

> To tinker this example code to work best for you, determine at what [analog value from the soil moisture readings](#test-your-soil-moisture-readings-on-your-pi) you want to water your plant, as your thirsty plant's average moisture reading might differ from our example value of `60000`.

Reviewer 4: ```suggestion
```

Now that we've removed the adafruit CLI section, I think it's safe to just delete this line entirely: we no longer provide users with guidance on adjusting this value.
